bundles/org.openhab.binding.irobot/pom.xml

```diff
@@ -13,4 +13,35 @@
   <artifactId>org.openhab.binding.irobot</artifactId>
 
   <name>openHAB Add-ons :: Bundles :: iRobot Binding</name>
+
+  <properties>
+    <bnd.importpackage>!org.apache.tapestry5.json.*,!org.codehaus.jettison.json.*,!org.json.*,!com.fasterxml.jackson.*</bnd.importpackage>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.jayway.jsonpath</groupId>
+      <artifactId>json-path</artifactId>
+      <version>2.5.0</version>
+      <scope>compile</scope>
+    </dependency>
```

Comment:
Actually, we try to keep dependencies to a minimum. Is there any reason for reading the JSON data by json-path instead of Gson?

Reply:
I refactor the implamentation now. The reason is, that current implementation support 980-Series roomba vaccums only. Next PR's will heavy use the jsonpath.

Reply:
Can you elaborate what the actual reason for moving to json-path is? Here is the list of default libraries which should preferably be used: https://www.openhab.org/docs/developer/guidelines.html#default-libraries Other libraries can also be used, but there needs to be a valid reason introducing new dependencies. OH often runs on hardware with limited resources. So, a major goal is to keep the application footprint small. This is also driven by keeping the code complexity of a software project of this scale in an acceptable range.

Reply:
Sure. Different robot models reply different json back. Since actual dto file is quite long and very hard to maintain, it's not really way to go, to extend the functionality. I've decided to use jsonpath as migration help. Additionally, there is no new dependency is introduced: jsonpath transform already use the library. Meanwhile, i found more elegant solution to do the task. As far this PR is in, i'll raise new one. Where some parts of this PR are already reworked.

Reply:
Frankly, I don't see how jsonpath makes things easier here. I understand that json could be your favorite approach for parsing JSON, but the code base of OH uses GSON. Using it consistently makes the code better maintainable and easier for others to make changes. The compile dependency is added to your bundle's JAR.